### PR TITLE
Add pre-commit check for dependencies hash in our workflows

### PR DESCRIPTION
**Pull request by amgebauer**
We need to manually set the dependencies hash in our workflows and keep them up to date. If we forget to keep all of them up to date, we might see the failure only after merge. This PR adds a simple check that all dependencies hashes are up to date in all workflows.

### CODE REVIEW

File: utilities/four_c_utils/src/four_c_utils/check_workflow_dependencies_hash.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import subprocess
+import os
+import yaml
+import sys
+
+
+def compute_dependencies_hash():
+    result = subprocess.run(
+        r"find dependencies docker -not -wholename '*/trilinos_develop/*' -not -name 'README.md' -type f -exec sha1sum {} \; | sort | sha1sum | cut -c -8",
```
Comment (sebproell) [inline utilities/four_c_utils/src/four_c_utils/check_workflow_dependencies_hash.py:16]: We are duplicating this line in a few places already. Can we extract it somewhere into a script?

Comment (amgebauer) [reply utilities/four_c_utils/src/four_c_utils/check_workflow_dependencies_hash.py:16]: True. I'll do it once we have synchronized with Gitlab.

**Comment by ppraegla** [comment]
@amgebauer 
>If we forget to keep all of them up to date, we might see the failure only after merge

Isn't this case already covered by the dependencies check in the Actions?

**Comment by amgebauer** [comment]
>>If we forget to keep all of them up to date, we might see the failure only after merge
>
>Isn't this case already covered by the dependencies check in the Actions?

yes, but if you don't update it in the nightly_tests workflow, you will only realize it after merge.

**Comment by ppraegla** [comment]
> > > If we forget to keep all of them up to date, we might see the failure only after merge
> > 
> > 
> > Isn't this case already covered by the dependencies check in the Actions?
> 
> yes, but if you don't update it in the nightly_tests workflow, you will only realize it after merge.

Yes, of course. I did not look at the new nightly tests yet. Thanks!

**Comment by amgebauer** [comment]
Close, in favor of #80 (I forgot that this one here is still open :see_no_evil:).

**Pull request closed without merging**